src/blocks/uptime.rs

```diff
@@ -0,0 +1,137 @@
+use std::time::Duration;
+use chan::Sender;
+
+use block::{Block, ConfigBlock};
+use config::Config;
+use de::deserialize_duration;
+use errors::*;
+use widgets::text::TextWidget;
+use widget::I3BarWidget;
+use input::I3BarEvent;
+use scheduler::Task;
+use std::fs::OpenOptions;
+use std::io::prelude::*;
+
+use uuid::Uuid;
+
+pub struct Uptime {
+    text: TextWidget,
+    id: String,
+    update_interval: Duration,
+
+    //useful, but optional
+    #[allow(dead_code)] config: Config,
+    #[allow(dead_code)] tx_update_request: Sender<Task>,
+}
+
+#[derive(Deserialize, Debug, Default, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct UptimeConfig {
+    /// Update interval in seconds
+    #[serde(default = "UptimeConfig::default_interval", deserialize_with = "deserialize_duration")]
+    pub interval: Duration,
+}
+
+impl UptimeConfig {
+    fn default_interval() -> Duration {
+        Duration::from_secs(60)
+    }
+}
+
+impl ConfigBlock for Uptime {
+    type Config = UptimeConfig;
+
+    fn new(block_config: Self::Config, config: Config, tx_update_request: Sender<Task>) -> Result<Self> {
+        Ok(Uptime {
+            id: Uuid::new_v4().simple().to_string(),
+            update_interval: block_config.interval,
+            text: TextWidget::new(config.clone()),
+            tx_update_request: tx_update_request,
+            config: config,
+        })
+    }
+}
+
+fn read_file(path: &str) -> Result<String> {
```

Comment:
I'm quite sure this is a duplicate. Would you mind finding the original and either putting it into a util file or accessing it instead?

Reply:
Can do, this is from the battery block.